packages/cli/templates/vue3/src/provideFieldPlugin.ts

```diff
@@ -0,0 +1,21 @@
+import { ref, provide } from 'vue'
+import { createFieldPlugin, FieldPluginResponse } from '@storyblok/field-plugin'
+
+let provided = false
+
+export const provideFieldPlugin = () => {
+  if (provided) {
+    throw new Error('Do not call `provideFieldPlugin()` more than once.')
+  }
+
+  provided = true
+  const plugin = ref<FieldPluginResponse>({
+    type: 'loading',
+  })
+  createFieldPlugin((newState) => {
+    plugin.value = newState
+  })
```

Comment:
Best practice to clean up side effects with [onUnmounted](https://vuejs.org/api/composition-api-lifecycle.html#onunmounted). I guess that also should execute `provided = false`?

Also probably best to wrap `createFieldPlugin` inside [onMounted](https://vuejs.org/api/composition-api-lifecycle.html#onmounted). Not sure if this has any benefit unless the component is server-rendered. The docs do not specify _why_ it should wrap side effects:

> This hook is typically used for performing side effects that need access to the component's rendered DOM

Reply:
Good point. I'm gonna clean it up with onUnmounted. I also tried using onMounted, but with that, we don't get `plugin` initially. So I'll just pass onMounted. If needed somehow, users will use it :)

<img width="662" alt="Screenshot 2023-04-13 at 10 47 23@2x" src="https://user-images.githubusercontent.com/499898/231706048-10c70700-5280-403f-94a0-5e4774683629.png">